app/src/main/java/org/wikipedia/donate/donationreminder/DonationReminderHelper.kt

```diff
@@ -143,6 +151,74 @@ object DonationReminderHelper {
             }
         }
     }
+
+    fun maybeShowSurveyDialog(activity: Activity) {
+        if (!isEnabled) return
+
+        val config = Prefs.donationReminderConfig
+        if (config.isSurveyShown) return
+
+        // when user sets up the reminder
+        val hasSetupReminder = config.donateAmount > 0 && config.articleFrequency > 0
+        if (hasSetupReminder) {
+            val userGroup = getUserGroup()
+            when (userGroup) {
+                "A" -> {
+                    // Group A: Show survey on next article visit after setting up reminder
+                    showFeedbackOptionsDialog(activity, Constants.InvokeSource.PAGE_ACTIVITY)
+                }
+                "B" -> {
+                    // Group B: Show survey on the next article visit after seeing reminder impressions two times
+                    if (config.finalPromptCount >= MAX_REMINDER_PROMPTS) {
+                        showFeedbackOptionsDialog(activity, Constants.InvokeSource.PAGE_ACTIVITY)
+                    }
+                }
+            }
+            return
+        }
+
+        // User has not taken any action on the initial prompt
+        // Show survey on next article visit if this continues for continuous 5 times
+        if (config.initialPromptCount >= MAX_INITIAL_REMINDER_PROMPTS) {
+            showFeedbackOptionsDialog(activity, Constants.InvokeSource.PAGE_ACTIVITY)
+            return
+        }
+    }
+
+    private fun getUserGroup(): String {
+        return if (Prefs.appInstallId.hashCode() % 2 == 0) "A" else "B"
+    }
+
+    private fun showFeedbackOptionsDialog(activity: Activity, invokeSource: Constants.InvokeSource) {
+        var dialog: AlertDialog? = null
+        val binding = DialogFeedbackOptionsBinding.inflate(activity.layoutInflater)
+        binding.titleText.text = activity.getString(R.string.donation_reminders_survey_dialog_title)
+        binding.messageText.text = activity.getString(R.string.donation_reminders_survey_dialog_message)
+        binding.feedbackInputContainer.isVisible = true
+        binding.feedbackInput.setOnFocusChangeListener { _, hasFocus ->
+            if (hasFocus) {
+                binding.dialogContainer.postDelayed({
+                    binding.dialogContainer.fullScroll(ScrollView.FOCUS_DOWN)
+                }, 200)
+            }
+        }
+
+        val clickListener = View.OnClickListener {
+            val feedbackOption = (it as TextView).text.toString()
+            dialog?.dismiss()
+        }
+        binding.optionSatisfied.setOnClickListener(clickListener)
+        binding.optionNeutral.setOnClickListener(clickListener)
+        binding.optionUnsatisfied.setOnClickListener(clickListener)
```

Comment:
Setting up click listeners for the checkboxes does not make sense to me. Would it be possible to remove the click listeners here and only submit the survey in the `submitButton` click listener?